src/Storages/StorageMergeTree.cpp

```diff
@@ -2482,4 +2482,12 @@ void StorageMergeTree::fillNewPartName(MutableDataPartPtr & part, DataPartsLock
     part->setName(part->getNewName(part->info));
 }
 
+void StorageMergeTree::fillNewPartNameAndResetLevel(MutableDataPartPtr & part, DataPartsLock &)
```

Comment:
Copy of `StorageMergeTree::fillNewPartName` with `level = 0`. The former is called in other places, so I am not sure it is safe to reset it there.